include/kodlab_mjbots_sdk/imu_data.h

```diff
@@ -0,0 +1,373 @@
+   * (default deg/s)
+   */
+  Eigen::Vector3<Scalar> ang_bias_uncertainty_ = {0, 0, 0};
+
+  /**
+   * @brief World frame rotation offset
+   * @note Useful to change the direction of gravity
+   */
+  Eigen::Quaternion<Scalar> world_offset_ = {1, 0, 0, 0};
+
+  /**
+   * @brief Invalidates cached data
+   */
+  void InvalidateCache()
+  {
+    euler_.invalidate();
+    rot_mat_.invalidate();
+  }
+
+public:
+  /**
+   * @brief Construct a new \c IMUData object.
+   */
+  IMUData() = default;
+
+  /**
+   * @brief Construct a new \c IMUData object with populated data.
+   *
+   * @param quat_in attitude quaternion
+   * @param ang_rate_in angular velocity about <code>[x, y, z]</code> axes
+   * @param accel_in linear acceleration along <code>[x, y, z]</code> axes
+   * @param world_offset_in world frame rotation offset (default identity)
+   */
+  IMUData(const Eigen::Quaternion<Scalar> &quat_in,
+          const Eigen::Vector3<Scalar> &ang_rate_in,
+          const Eigen::Vector3<Scalar> &accel_in,
+          const Eigen::Quaternion<Scalar> &world_offset_in = {1, 0, 0, 0})
+      : quat_raw_(quat_in),
+        quat_(world_offset_in * quat_in),
+        ang_rate_(ang_rate_in),
+        accel_(accel_in),
+        world_offset_(world_offset_in) {}
+
+  /**
+   * @brief Construct a new \c IMUData object with populated data, biases,
+   *        and uncertainty.
+   *
+   * @param quat_in attitude quaternion
+   * @param ang_rate_in angular velocity about <code>[x, y, z]</code> axes
+   * @param accel_in linear acceleration along <code>[x, y, z]</code> axes
+   * @param ang_rate_bias_in gyroscope bias about <code>[x, y, z]</code> axes
+   * @param att_uncertainty_in attitude uncertainty quaternion
+   * @param ang_bias_uncertainty_in gyroscope bias uncertainty about
+   *                                <code>[x, y, z]</code> axes
+   * @param world_offset_in world frame rotation offset (default identity)
+   */
+  IMUData(const Eigen::Quaternion<Scalar> &quat_in,
+          const Eigen::Vector3<Scalar> &ang_rate_in,
+          const Eigen::Vector3<Scalar> &accel_in,
+          const Eigen::Vector3<Scalar> &ang_rate_bias_in,
+          const Eigen::Quaternion<Scalar> &att_uncertainty_in,
+          const Eigen::Vector3<Scalar> &ang_bias_uncertainty_in,
+          const Eigen::Quaternion<Scalar> &world_offset_in = {1, 0, 0, 0})
+      : quat_raw_(quat_in),
+        quat_(world_offset_in * quat_in),
+        ang_rate_(ang_rate_in),
+        accel_(accel_in),
+        ang_rate_bias_(ang_rate_bias_in),
+        att_uncertainty_(att_uncertainty_in),
+        ang_bias_uncertainty_(ang_bias_uncertainty_in),
+        world_offset_(world_offset_in) {}
+
+  /**
+   * @brief Construct a new \c IMUData object from a
+   *        \c mjbots::pi3hat::IMUData object
+   *
+   * @param pi_att attitude object
+   * @param world_offset_in world frame rotation offset (default identity)
+   */
+  IMUData(const ::mjbots::pi3hat::Attitude &pi_att,
```

Comment:
This should be fixed in #29 along with [above](https://github.com/KodlabPenn/kodlab_mjbots_sdk/pull/16/files#r935106685)